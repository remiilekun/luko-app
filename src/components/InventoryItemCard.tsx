import React from "react";
import {
  Image,
  StyleSheet,
  Text,
  TouchableOpacity,
  TouchableOpacityProps,
  View,
  ViewStyle,
} from "react-native";
import { InventoryItem } from "../navigation/types";
import { colors } from "../theme/colors";
import { fonts } from "../theme/fonts";

interface InventoryItemCardProps extends TouchableOpacityProps {
  item: InventoryItem;
  style?: ViewStyle;
}

const InventoryItemCard = ({
  item,
  style,
  ...rest
}: InventoryItemCardProps): JSX.Element => {
  return (
    <TouchableOpacity
      activeOpacity={0.8}
      style={[styles.wrapper, style]}
      {...rest}
    >
      <Image style={styles.image} source={{ uri: item.photo }} />
      <View style={styles.content}>
        <Text testID="inventory-card-title" style={styles.name}>
          {item.name}
        </Text>
        <Text testID="inventory-card-price" style={styles.price}>
          {item.value}
        </Text>
      </View>
    </TouchableOpacity>
  );
};

export default InventoryItemCard;

const styles = StyleSheet.create({
  wrapper: {
    backgroundColor: colors.white,
    borderRadius: 14,
    overflow: "hidden",
    shadowColor: "#06080D",
    shadowOffset: { height: 10, width: 0 },
    shadowOpacity: 0.3,
    shadowRadius: 20,
  },
  image: {
    width: "100%",
    height: 150,
  },
  content: {
    paddingHorizontal: 20,
    paddingVertical: 15,
  },
  name: {
    fontFamily: fonts.bold,
    color: colors.darkGrey,
    fontSize: 19,
    lineHeight: 26,
    marginBottom: 30,
  },
  price: {
    fontFamily: fonts.regular,
    color: colors.darkGrey2,
    fontSize: 15,
    lineHeight: 20,
  },
});
