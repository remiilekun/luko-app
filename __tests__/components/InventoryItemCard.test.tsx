import "react-native";
import React from "react";
import renderer from "react-test-renderer";
import InventoryItemCard from "../../src/components/InventoryItemCard";

const item = {
  id: "1",
  name: "Glasses",
  value: "800",
};

describe("components", () => {
  describe("InventoryItemCard", () => {
    it("renders correctly", () => {
      const tree = renderer.create(<InventoryItemCard item={item} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
