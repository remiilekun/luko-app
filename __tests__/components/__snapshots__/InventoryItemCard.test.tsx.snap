// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components InventoryItemCard renders correctly 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderRadius": 14,
      "opacity": 1,
      "overflow": "hidden",
      "shadowColor": "#06080D",
      "shadowOffset": Object {
        "height": 10,
        "width": 0,
      },
      "shadowOpacity": 0.3,
      "shadowRadius": 20,
    }
  }
>
  <Image
    source={
      Object {
        "uri": undefined,
      }
    }
    style={
      Object {
        "height": 150,
        "width": "100%",
      }
    }
  />
  <View
    style={
      Object {
        "paddingHorizontal": 20,
        "paddingVertical": 15,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#2C2302",
          "fontFamily": "Circular-Bold",
          "fontSize": 19,
          "lineHeight": 26,
          "marginBottom": 30,
        }
      }
      testID="inventory-card-title"
    >
      Glasses
    </Text>
    <Text
      style={
        Object {
          "color": "#6B6651",
          "fontFamily": "Circular-Regular",
          "fontSize": 15,
          "lineHeight": 20,
        }
      }
      testID="inventory-card-price"
    >
      800
    </Text>
  </View>
</View>
`;
